core/web/router.go

```diff
@@ -150,8 +150,20 @@ func sessionAuth(store *store.Store, c *gin.Context) error {
 		return ErrorAuthFailed
 	}
 
-	_, err := store.AuthorizedUserWithSession(sessionID)
-	return err
+	user, err := store.AuthorizedUserWithSession(sessionID)
+	if err != nil {
+		return err
+	}
+	c.Set("user", &user)
+	return nil
+}
+
+func authenticatedUser(c *gin.Context) (*models.User, bool) {
+	obj, ok := c.Get("user")
```

Comment:
Might be helpful to put this in a constant with a descriptive name, e.g: SESSION_KEY_USER